### PR TITLE
Add date statistics to Project Tasks

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

File: project_task_statistic/README.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
```
Comment (yvaucher) [inline project_task_statistic/README.rst]: Please add title

File: project_task_statistic/README.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Adds a new tab Statistics to the task form, similar to the Issue statistics.
+It could be used for information alone, to use with Automated Actions, or to
+for SLA control.
+
```
Comment (yvaucher) [inline project_task_statistic/README.rst]: And bug tracking

File: project_task_statistic/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Project Task Statistic',
+    'summary': 'Add a new tab Statistics to the task form',
+    'version': '1.0',
```
Comment (yvaucher) [inline project_task_statistic/__openerp__.py]: missing license key

File: project_task_statistic/project_task_sla_close.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+class project_task_close_action(models.Model):
+    _inherit = "project.task"
+
+    date_opened = fields.Datetime('Date Opened')
+    date_closed = fields.Datetime('Date Closed')
+
+    @api.model
+    def create(self, vals):
+        if not (self.date_opened) and self.stage_id.state == 'open':
+            vals['date_opened'] = fields.datetime.now().strftime(
```
Comment (yvaucher) [inline project_task_statistic/project_task_sla_close.py]: please use `fields.Datetime` `now()` methods
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.fields.Datetime

File: project_task_statistic/project_task_sla_close.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                '%Y-%m-%d %H:%M:%S')
+
+        elif self.stage_id.state in ('cancelled', 'done'):
+            vals['date_closed'] = fields.datetime.now().strftime(
```
Comment (yvaucher) [inline project_task_statistic/project_task_sla_close.py]: `fields.Datetime.now()` to use

File: project_task_statistic/project_task_sla_close.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        else:
+            vals['date_closed'] = None
+        res = super(project_task_close_action, self).create(vals)
+        return res
+
+    @api.one
+    def write(self, vals):
+        if not (self.date_opened) and self.stage_id.state == 'open':
+            vals['date_opened'] = fields.datetime.now().strftime(
+                '%Y-%m-%d %H:%M:%S')
```
Comment (yvaucher) [inline project_task_statistic/project_task_sla_close.py]: `fields.Datetime.now()` to use

File: project_task_statistic/project_task_sla_close.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+        elif self.stage_id.state in ('cancelled', 'done'):
+            if not self.date_closed:
+                vals['date_closed'] = fields.datetime.now(
+                    ).strftime('%Y-%m-%d %H:%M:%S')
```
Comment (yvaucher) [inline project_task_statistic/project_task_sla_close.py]: `fields.Datetime.now()` to use

**Comment by rafaelbn** [comment]
@dreispt Let me know when ready for testing, thanks!

File: project_task_statistic/README.rst [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+Currently only the Starting and Closing dates are stored.
+Should also compute the days and working hours to Open and to Close.
+
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (yvaucher) [inline project_task_statistic/README.rst]: s/{//
s/_repo}//

File: project_task_statistic/project_task_model.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    @api.model
+    def create(self, vals):
+        task = super(Task, self).create(vals)
+        task._calc_statistics()
+        return task
+
+    @api.one
+    def write(self, vals):
+        task = super(Task, self).write(vals)
+        task._calc_statistics()
```
Comment (yvaucher) [inline project_task_statistic/project_task_model.py]:
```
'list' object has no attribute '_calc_statistic'
```

You might want to call _calc_statistics on `self`

Comment (dreispt) [reply project_task_statistic/project_task_model.py]: `task` is a record, not a list.
The calculation should be done on the record resulting from the write operation.
That error is coming from the `project_task_analytic_partner` module; I'll need to check that.

Comment (yvaucher) [reply project_task_statistic/project_task_model.py]: oh wait.

`@api.one` ?

You are returning a list of record set.

**Comment by LoisRForgeFlow** [comment]
Is this dead  :skull: (and should be closed) or alive  :dancer: (and should be reviewed)? @dreispt

**Comment by dreispt** [comment]
I can revive it.

**Comment by andhit-r** [comment]
Hi @dreispt , i continue your work on #375 . Please let me know if you prefer me to create PR on your branch.

**Comment by pedrobaeza** [comment]
Closing for now.

**Pull request closed without merging**